src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/freestyle/FreeStylePostBuildStep.java

```diff
@@ -75,9 +75,9 @@ public boolean perform(
 
         final FreestyleBuildInfoRequest request =
                 new FreestyleBuildInfoRequest(this.getSite(), this.branch, build);
-        JiraSenderFactory.getInstance(pipelineLogger)
+        new JiraSenderFactory(pipelineLogger)
```

Comment:
The `getInstance()` meant that we were sharing the same `JiraSenderFactory` instance for different cases. Since I made the `JiraSenderFactory` stateful by adding a `PipelineLogger`, we would also share the `PipelineLogger`. But `PipelineLoggers` for different Pipelines have different underlying `PrintStream`s, so this led to log events getting lost because they were sent to the wrong `PrintStream`. 

From a performance perspective it doesn't really matter that we're creating a new `JiraSenderFactory` in every case, because it's not being done in a high frequency, anyways.